pulp_2to3_migration/app/migration.py

```diff
@@ -238,14 +239,15 @@ def create_repo_version(migrator, pulp3_repo_name, pulp2_repo, pulp3_remote=None
             pulp3_repo_name(str): repository name in Pulp 3
             pulp2_repo(Pulp2Repository): a pre-migrated repository to create a repo version for
             pulp3_remote(remote): a pulp3 remote
+            repo_setup(boolean) indicated weather repo migration has a special setup
         """
 
         # Add a remote to every repo, even the migrated one, because remotes are migrated on
         # every run
         if pulp3_remote:
             pulp2_repo.pulp3_repository_remote = pulp3_remote
         # pulp2importer might not be migrated, e.g. config was empty
-        elif hasattr(pulp2_repo, 'pulp2importer'):
+        elif hasattr(pulp2_repo, 'pulp2importer') and not repo_setup:
```

Comment:
@goosemania  @jlsherrill correct my understanding if it is not correct:

- we create Remote if we do not have any special setup for the migrated repos, i.e. per each repo migrate repo and its native importer
- we create Remote if it was specified in pulp2_importer_repo_id
- in case we have special setup for repos and pulp2_importer_repo_id is not part of MP pulp3_remote will be null

Reply:
that sounds right to me

Reply:
```
{
    "plan": {
        "plugins": [
            {
                "repositories": [
                    {
                        "name": "foo1",
                        "repository_versions": [
                            {
                                "pulp2_distributor_repository_ids": [
                                    "file2"
                                ],
                                "pulp2_repository_id": "file"
                            }
                        ]
                    }
                ],
                "type": "iso"
            }
        ]
    },
    "pulp_created": "2020-01-28T16:13:00.625495Z",
    "pulp_href": "/pulp/api/v3/migration-plans/3acbfa86-77b9-4282-994e-0af025d7f681/"
}
$ http GET :24817/pulp/api/v3/pulp2repositories/
HTTP/1.1 200 OK
Allow: GET, HEAD, OPTIONS
Connection: close
Content-Length: 636
Content-Type: application/json
Date: Tue, 28 Jan 2020 16:13:43 GMT
Server: gunicorn/20.0.4
Vary: Accept, Cookie
X-Frame-Options: SAMEORIGIN

{
    "count": 1,
    "next": null,
    "previous": null,
    "results": [
        {
            "is_migrated": true,
            "not_in_plan": false,
            "pulp2_object_id": "5e1c932dc998ac367a9b79d0",
            "pulp2_repo_id": "file",
            "pulp3_distribution_hrefs": [
                "/pulp/api/v3/distributions/file/file/48d2845f-4e21-4757-9a90-faf06602e836/"
            ],
            "pulp3_publication_href": [
                "/pulp/api/v3/publications/file/file/fbd395ab-9f04-44c6-a657-e823f4cb2146/"
            ],
            "pulp3_remote_href": null,
            "pulp3_repository_version": "/pulp/api/v3/repositories/file/file/5198f10f-579f-4beb-b57b-73f0276e0d89/versions/1/",
            "pulp_created": "2020-01-28T16:13:08.808374Z",
            "pulp_href": "/pulp/api/v3/pulp2repositories/7f5ff806-8527-4a63-9c0e-e8e030400da3/"
        }
    ]
}
```

Reply:
EDIT for the cases "we create Remote " happens only when he importer has feed.